src/SFA.DAS.LevyTransferMatching.Web/Orchestrators/PledgeOrchestrator.cs

```diff
@@ -202,9 +202,38 @@ public async Task<ApplicationApprovedViewModel> GetApplicationApprovedViewModel(
             };
         }
         
-        public async Task<Dictionary<int, string>> ValidateLocations(LocationPostRequest request)
+        public async Task<LocationSelectViewModel> GetLocationSelectViewModel(LocationSelectRequest request)
         {
-            return await _validatorService.ValidateLocations(request);
+            var cacheItem = await RetrievePledgeCacheItem(request.CacheKey);
+
+            var locationSelectionGroups = cacheItem.MultipleValidLocations
```

Comment:
It's a bit hard to follow what this is doing

Reply:
I've refactored the LINQ statement a little in an attempt to improve code readability.